src/main/java/joules/Ui.java

```diff
@@ -0,0 +1,134 @@
+                + "      / .  \\;  \\  '_ /  | :.'  '_  | |\\  '_ '`) . (_ o _)  |(_ o _).    \n"
+                + " ___  |-'`| |  _`,/ \\ _/  |'   ( \\.-.| > (_)  ) |  (_,_)___| (_,_). '.  \n"
+                + "|   | |   ' : (  '\\_/ \\   ;' (`. _` /|(  .  .-' '  \\   .---..---.  \\  : \n"
+                + "|   `-'  /   \\ `\"/  \\  ) / | (_ (_) _) `-'`-'|___\\  `-'    /\\    `-'  | \n"
+                + " \\      /     '. \\_/``\".'   \\ /  . \\ /  |        \\\\       /  \\       /  \n"
+                + "  `-..-'        '-----'      ``-'`-''   `--------` `'-..-'    `-...-'   \n");
+        System.out.println(" What can I do for you?");
+    }
+
+    /**
+     * Displays the goodbye message and ASCII art when the chatbot exits.
+     */
+    public void showGoodbye() {
+        System.out.println(" _______      ____     __   .-''-.  .---.  \n"
+                + "\\  ____  \\    \\   \\   /  /.'_ _   \\ \\   /  \n"
+                + "| |    \\ |     \\  _. /  '/ ( ` )   '|   |  \n"
+                + "| |____/ /      _( )_ .'. (_ o _)  | \\ /   \n"
+                + "|   _ _ '.  ___(_ o _)' |  (_,_)___|  v    \n"
+                + "|  ( ' )  \\|   |(_,_)'  '  \\   .---. _ _   \n"
+                + "| (_{;}_) ||   `-'  /    \\  `-'    /(_I_)  \n"
+                + "|  (_,_)  / \\      /      \\       /(_(=)_) \n"
+                + "/_______.'   `-..-'        `'-..-'  (_I_)\n");
+        System.out.println(" Hope to see you again soon!");
+    }
+
+    /**
+     * Reads a line of input from the user.
+     *
+     * @return The user's input as a {@link String}.
+     */
+    public String readInput() {
+        return Ui.SCANNER.nextLine();
+    }
+
+    /**
+     * Lists all tasks in the provided {@link TaskList}.
+     *
+     * @param tasks The {@link TaskList} containing tasks to display.
+     */
+    public void listTasks(TaskList tasks) {
+        System.out.println(" You got this! These are your tasks:");
+        tasks.printTaskList();
+    }
+
+    /**
+     * Displays a message that the specified task has been marked as done.
+     *
+     * @param t The task that was marked.
+     */
+    public void markTask(Task t) {
+        System.out.println(" Keep up the good work! I've marked this task as done:");
+        System.out.println("   " + t);
+    }
+
+    /**
+     * Prints all tasks from the given TaskList that contain the specified keyword
+     * in their description.
+     *
+     * @param keyword The string to search for in each task's description.
+     * @param tasks The TaskList containing the tasks to search through.
+     */
+    public void listMatchingTasks(String keyword, TaskList tasks) {
+        System.out.println(" I have found these matching tasks:");
+        tasks.printMatchingTaskList(keyword);
+    }
+
+    /**
+     * Displays a message that the specified task has been unmarked (not done).
+     *
+     * @param t The task that was unmarked.
+     */
+    public void unMarkTask(Task t) {
```

Comment:
Can refer to the comment in Joules.java Line 69 on possible appropriate naming
```suggestion
    public void unmarkTask(Task t) {
```